ANMN/burst_averaged_product/burst_average.py

```diff
@@ -0,0 +1,414 @@
+    output_netcdf_obj.geospatial_lon_max = input_netcdf_obj['LONGITUDE'][:].max()
+
+    depth_burst_mean_val = burst_vars['DEPTH']['var_mean']
+    output_netcdf_obj.geospatial_vertical_min = (np.ma.masked_array(depth_burst_mean_val, np.isnan(depth_burst_mean_val))).min()
+    output_netcdf_obj.geospatial_vertical_max = (np.ma.masked_array(depth_burst_mean_val, np.isnan(depth_burst_mean_val))).max()
+
+    time_burst_val_dateobj = num2date(time_burst_vals, input_netcdf_obj['TIME'].units, input_netcdf_obj['TIME'].calendar)
+    output_netcdf_obj.time_coverage_start = time_burst_val_dateobj.min().strftime('%Y-%m-%dT%H:%M:%SZ')
+    output_netcdf_obj.time_coverage_end   = time_burst_val_dateobj.max().strftime('%Y-%m-%dT%H:%M:%SZ')
+
+    # set up dimensions
+    output_netcdf_obj.createDimension("TIME", len(time_burst_vals))
+
+    # set up variables
+    var_time = output_netcdf_obj.createVariable("TIME", "d", ("TIME",), fill_value=get_imos_parameter_info('TIME', '_FillValue'))
+    var_lat  = output_netcdf_obj.createVariable("LATITUDE", "d", fill_value=get_imos_parameter_info('LATITUDE', '_FillValue'))
+    var_lon  = output_netcdf_obj.createVariable("LONGITUDE", "d", fill_value=get_imos_parameter_info('LONGITUDE', '_FillValue'))
+
+    for var in burst_vars.keys():
+        fillvalue = get_imos_parameter_info(var, '_FillValue')
+        if fillvalue == []:
+            fillvalue = 999999.0
+
+        output_var_mean    = output_netcdf_obj.createVariable(var, "f4", ("TIME",), fill_value=fillvalue)
+        output_var_min     = output_netcdf_obj.createVariable('%s_burst_min' % var, "f4", ("TIME",), fill_value=fillvalue)
+        output_var_max     = output_netcdf_obj.createVariable('%s_burst_max' % var, "f4", ("TIME",), fill_value=fillvalue)
+        output_var_sd      = output_netcdf_obj.createVariable('%s_burst_sd' % var, "f4", ("TIME",), fill_value=fillvalue)
+        output_var_num_obs = output_netcdf_obj.createVariable('%s_num_obs' % var, "i4", ("TIME",), fill_value=fillvalue)
+
+        # set up 'bonus' var att from original FV01 file into FV02
+        input_var_object   = input_netcdf_obj[var]
+        input_var_list_att = input_var_object.__dict__.keys()
+        var_att_disposable = ['name', 'long_name', 'valid_min', 'valid_max', \
+                              '_FillValue', 'ancillary_variables', \
+                              'axis', 'ChunkSize', 'coordinates']
+        for var_att in [att for att in input_var_list_att if att not in var_att_disposable]:
+            setattr(output_netcdf_obj[var], var_att, getattr(input_netcdf_obj[var], var_att))
+            if var_att != 'comment':
+                setattr(output_var_min, var_att, getattr(input_netcdf_obj[var], var_att))
+                setattr(output_var_max, var_att, getattr(input_netcdf_obj[var], var_att))
+                setattr(output_var_sd, var_att, getattr(input_netcdf_obj[var], var_att))
+
+        var_units = get_imos_parameter_info(var, 'units')
```

Comment:
That's fine, you shouldn't be anticipating for problems (and possibly introduce new ones). The input file should be passing the checker. As soon as it does then the product will pass it too.